DAQmxBaseApp/src/drvDaqMxBase.c

```diff
@@ -3960,14 +4074,23 @@ static void daqThread(void *param)
                     /*asynPrint(pPvt->pasynUser, ASYN_TRACE_FLOW,
                       "sending IO/intr (signal=%d).\n",signal);*/
                     if (signal >= 0 && signal < pPvt->nChannels) {
+                        int i;
+                        epicsFloat64 value = *(epicsFloat64*)pPvt->aioPvt[signal]->data;
+                        if (pPvt->samplesRead > 0) {
+                            value = 0.0;
+                            for(i = 0; i<pPvt->samplesRead; ++i) {
+                                value += ((epicsFloat64*)pPvt->aioPvt[signal]->data)[i];
+                            }
+                            value /= (epicsFloat64)pPvt->samplesRead;
+                        }
```

Comment:
Looks very similar to lines #4219-4227.
Possible convert to "templated function"?

Reply:
May lead to other incompatibilities, so decided not to refactor.